scripts/e2e/run-e2e.sh

```diff
@@ -33,4 +33,4 @@ kubectl create namespace ${E2E_PROJECT_NAME} --dry-run=client -o yaml | kubectl
 cd ../../python/sdk
 pip install pipenv==2023.7.23
 pipenv install --dev --skip-lock --python ${PYTHON_VERSION}
-pipenv run pytest -n=8 -W=ignore --cov=merlin -m "not (gpu or feast or batch or pyfunc or local_server_test or cli or customtransformer)" --durations=0
+pipenv run pytest -n=4 -W=ignore --cov=merlin -m "not (gpu or feast or batch or pyfunc or local_server_test or cli or customtransformer)" --durations=0
```

Comment:
Why do we need to change the number of e2e test workers? :O
/

Reply:
Related to your comment [here](https://github.com/caraml-dev/merlin/pull/471#discussion_r1363564013), where I increased default_deployment_config so reducing the number of test worker will prevent from exceeding the test cluster resource.

Reply:
Ah oops, got it, thanks for the explanation!